Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -97,8 +101,93 @@ internal SFSession GetSession(string connStr)
                     }
                 }
             }
-            return null;
+            return NewSession(connStr, password);
         }
+        
+        internal Task<SFSession> GetSessionAsync(string connStr, SecureString password, CancellationToken cancellationToken)
+        {
+            s_logger.Debug("SessionPool::GetSession");
+            if (!_pooling)
+                return NewSessionAsync(connStr, password, cancellationToken);
+            lock (s_sessionPoolLock)
```

Comment:
GetSession and GetSessionAsync have a very large common part. Please extract it to a separate method.

Reply:
thx